xsuite/src/cli/newWalletCmd.ts

```diff
@@ -30,26 +33,28 @@ const action = async ({
   let keystore: Keystore;
   if (password === undefined) {
     try {
-      keystore = await Keystore.createFile(walletPath);
+      keystore = await Keystore.createFile(walletPath, pemPath);
       log();
     } catch (err: any) {
       logError(err.message);
       return;
     }
   } else {
-    keystore = Keystore.createFile_unsafe(walletPath, password);
+    keystore = Keystore.createFile_unsafe(walletPath, password, pemPath);
   }
   logSuccess(`Wallet created at "${walletPath}".`);
   log();
   log(chalk.bold.blue("Address:") + ` ${keystore.newSigner()}`);
-  log();
-  log(chalk.bold.blue("Private key:"));
-  log(
-    keystore
-      .getMnemonicWords()
-      .map((w, i) => `  ${i + 1}. ${w}`)
-      .join("\n"),
-  );
-  log();
-  log(chalk.bold.yellow(`Please backup the private key in a secure place.`));
+  if (!pemPath) {
```

Comment:
It would be best if this is inferred from the keystore created instead of from the command parameters. The keystore class could have a `kind` property.

Reply:
👍